packages/core/src/factory.ts

```diff
@@ -33,101 +35,124 @@ export function factory<Resource, Input, Output, Config, LoadOpts>(
         }
       },
 
-      process: async (input: Input, c?: Config) => {
+      process: (input: Input, c?: Config) => {
```

Comment:
You could keep `async` here, you are calling `Promise.resolve` at the end anyway (`components.router.route()` is also async). Without `async` someone could think that is dealing with sync block.

Reply:
Fixed in d19d15e